backend/WebApplication1/Infrastructure/Debugging/FileLogger.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.IO;
+using WebApplication1.Infrastructure.Extensions;
+using WebApplication1.Infrastructure.Debugging.Abstract;
+using WebApplication1.Models;
+
+namespace WebApplication1.Infrastructure.Debugging
+{
+    public class FileLogger : ILogger, IDisposable
+    {
+        private static readonly string _mainLogFileName = AppSettings.Configuration.MainLogFileName;
+        private Lazy<FileLogger> _mainLogger = new Lazy<FileLogger>(() => new FileLogger(_mainLogFileName));
```

Comment:
Has this been tested? Where is the file being saved? It's not created in dependency injection hook

Reply:
It's not supposed to be injected. We're creating it here - notice the "new" keyword.
File is saved in the root directory (we should probably note this down and fix it eventually(so our root wouldn't be horribly messed up)).

Reply:
FileLogger is being injected in Startup.cs
`services.AddScoped<ILogger, FileLogger>(logger => new FileLogger($"Logs/log_{Environment.TickCount.ToString()}.log"));`
And the Logs/log is being created. In mainlog case it's created there, so I'll recheck if it's created in root directory